### PR TITLE
Jesse Reichel – Sprint-Challenge-Authentication

### PR DESCRIPTION
**Pull request by frankfaustino**

### CODE REVIEW

File: ANSWERS.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 1.  Describe Middleware, Sessions (as we know them in express), bcrypt and JWT.
```
Comment (lefrenk) [inline ANSWERS.md]: Well done! 👌

File: api/models/userModels.js [hunk 2]
```diff
@@ -5,19 +5,40 @@ const Schema = mongoose.Schema;
...
+      next(error)
+    })
   // https://github.com/kelektiv/node.bcrypt.js#usage
   // Fill this middleware in with the Proper password encrypting, bcrypt.hash()
   // if there is an error here you'll need to handle it by calling next(err);
   // Once the password is encrypted, call next() so that your userController and create a user
 });
 
 UserSchema.methods.checkPassword = function(plainTextPW, callBack) {
+  bcrypt.compare(plainTextPW, this.password)
```
Comment (lefrenk) [inline api/models/userModels.js]: 👌

File: api/controllers/user.js [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
 
-const createUser = (req, res) => {
-  const { username, password } = req.body;
+const createUser = (req, res, error) => {
+  // const { username, password } = req.body;
+  console.log(req.body)
+  const newUser = new User(req.body)
+    .save()
+    .then((user) => {
+      res.status(200).json(user)
```
Comment (lefrenk) [inline api/controllers/user.js]: Status code should be 201 (https://httpstatuses.com/201). Also, we probably don't want to be sending the hashed password to the client. 😜

File: api/controllers/login.js [hunk 4]
```diff
@@ -15,16 +15,20 @@ const login = (req, res) => {
     }
     user.checkPassword(password, (nonMatch, hashMatch) => {
       // This is an example of using our User.method from our model.
-      if (nonMatch !== null) {
-        res.status(422).json({ error: 'passwords dont match' });
-        return;
-      }
+      // if (nonMatch === null) {
```
Comment (lefrenk) [inline api/controllers/login.js]: Nice catch! As you've correctly inferred, `hashMatch` is equivalent to the `response` from `bcrypt.compare` matching the plain text password and the hashed password. We just need to handle the `error` as well:

```
if (error) return res.status(500).json({ error: error.message })
```

https://www.npmjs.com/package/bcrypt#to-check-a-password

**Comment by lefrenk** [comment]
Well done, Jesse! Everything looks good, your code is clean and concise. You're doing great, keep it up! ☺️

File: api/controllers/user.js (hunk 3 above)
Comment (KryoKorpz) [reply api/controllers/user.js]: sssshhhhh they like it ;)

**Pull request closed without merging**